pep-0692.rst

```diff
@@ -0,0 +1,344 @@
+
+Assignment
+----------
+A function typed using the ``**kwargs: **Movie`` construct can be assigned to another callable type only if they are compatible. This can happen in the following cases:
+
+1. Both destination and source have a ``**kwargs: **TypedDict`` parameter, the destination ``TypedDict`` is assignable to the source ``TypedDict`` and the rest of the parameters are the compatible.
+
+2. If the destination callable contains traditionally typed ``**kwargs: T`` and the source callable is typed using ``**kwargs: **TypedDict`` then each ``TypedDict`` value has to be assignable to type ``T``. Again, the rest of the parameters has to be compatible.
+
+3. If the destination callable doesn't contain ``**kwargs`` and the source callable contains ``**kwargs: **TypedDict`` then all of the ``TypedDict`` fields should be assignable to individual named parameters. Again, the rest of the parameters has to be compatible. For example:
+
+::
+
+    class MovieBase(TypedDict, total=True):
+        title: str
+    
+    class Movie(MovieBase, total=False):
+        year: int
+    
+    def foo(**kwargs: **Movie): ...
+    def bar(title: str, year: int = ...): ...
+
+    bar = foo  # OK!
+
+4. If the destination callable contains ``**kwargs: **TypedDict`` and the source callable containes either untyped or traditionally typed ``**kwargs: T`` then an error should be raised.
+
+5. If the destination callable contains ``**kwargs: **TypedDict`` then source callable should have keyword parameters assignable to the key value pairs in the ``TypedDict``. Again, the rest of the parameters should be compatible.
+
+``TypedDict`` unions
+--------------------
+
+It is possible to create unions of typed dictionaries. However, supporting typing ``**kwargs`` with a union of typed dicts would greatly increase the complexity of the implementation of this PEP and there seems to be no compelling use case to justify the support for this. Therefore, using unions of typed dictionaries to type ``**kwargs`` as described in the context of this PEP should result in an error:
+
+::
+
+    class Book(TypedDict):
+        genre: str
+        pages: int
+    
+    TypedDictUnion = Movie | Book
+
+    def foo(**kwargs: **TypedDictUnion) -> None: ...  # WRONG! Cannot use a union of TypedDicts to type **kwargs
+
+
+Additional keys
+---------------
+
+A big advantage of using ``**kwargs`` is the ability to pass arbitrary dictionaries containing keyword-value pairs to the function and let the function pick and use only the ones it needs. However, currently, there is no way to define a ``TypedDict`` that would allow additional keys, so supporting this feature would require an addition to the ``TypedDict`` specification. If this limitation blocks the acceptance of this PEP, then the PEP's scope would have to be extended or a new PEP proposing additional keys for ``TypedDict`` would have to be proposed.
```

Comment:
This section seems speculative, I would remove personally, or add as a limitation / rejected idea. It shouldn't be in Specification.

Reply:
Agree